dev-docs/security-overview.md

```diff
@@ -0,0 +1,417 @@
+
+✅ **Verification ensures:**
+
+- The actual deployed state matches the expected trusted state.
+- Root CA and Mesh CA certificates of the service mesh are provided securely to workload operators and data owners.
+
+## Attestation scope
+
+This section provides a concise overview of key components whose integrity is verified—either directly or indirectly—through remote attestation:
+
+### Launch measurement
+
+The CPU generates a hardware-enforced cryptographic hash of the files initially loaded into memory, known as the launch measurement. This measurement guarantees the integrity of the following components:
+
+- **Kernel and kernel command line**
+- **Minimal init filesystem (`initramfs`)**
+- **Root filesystem:** Verified indirectly since the kernel includes `dm-verity` drivers, and the root filesystem’s reference hash is provided as part of the kernel command line.
+- **`kata-agent`:** Located within the verified root filesystem.
+
+### `HOSTDATA`
+
+In addition, the CPU integrates the `HOSTDATA` field into the attestation report:
+
+- `HOSTDATA` is a hardware-enforced field always included in attestation reports.
+- It holds the cryptographic hash of the runtime policy enforced by the `kata-agent`.
+
+## Secret recovery
+
+When initially setting the manifest for the coordinator via the Contrast CLI, the coordinator returns a randomly generated recovery seed.
+The CLI encrypts this seed client-side using the private key of the seed-share owner.
+
+### What's derived from the recovery seed?
+
+- **Manifest Signing Key**: Used to ensure integrity and authenticity of the manifest stored on untrusted persistent volumes.
+- **Workload Secret Seeds**: Distributed securely to attested workloads, enabling them to derive encryption keys for securely storing sensitive data (for example secret keys) on untrusted persistent volumes.
+
+### Why is the recovery seed important?
+
+- ✅ It enables the secure restoration of sensitive data stored in persistent volumes.
+- ✅ Since the seed provides access to encrypted secrets, it must remain strictly confidential.
+
+## Example practical workflow
+
+You can follow the [Confidential emoji voting tutorial](https://docs.edgeless.systems/contrast/examples/emojivoto) as an example of a practical deployment workflow.
+
+## Supported Platforms
+
+### Intel TDX
+
+The architecture and considerations outlined in this document remain valid for Intel TDX deployments, with some terminology changes:
+
+- Intel TDX uses the `MRCONFIGID` field, which is equivalent to AMD SEV-SNP’s `HOSTDATA`.
+
+### Azure Kubernetes Service (AKS)
+
+Contrast can also be deployed on managed Kubernetes services, such as AKS. In this scenario, Kubernetes nodes run as virtual machines instead of bare-metal servers, necessitating nested virtualization. Therefore, Confidential Virtual Machines (CVMs) are launched within these node VMs.
```

Comment:
```suggestion
Contrast can also be deployed on Azure Kubernetes Service (AKS). In this scenario, Kubernetes nodes run as virtual machines instead of bare-metal servers, necessitating nested virtualization. Therefore, Confidential Virtual Machines (CVMs) are launched within these node VMs.
```
There is no roadmap for nested virt support in other managed k8s.